sdk/communication/azure-communication-chat/azure/communication/chat/_shared/user_credential_async.py

```diff
@@ -82,12 +83,14 @@ def _wait_till_inprogress_thread_finish_refreshing(self):
         self._lock.acquire()
 
     def _token_expiring(self):
-        return self._token.expires_on - self._get_utc_now() <\
-            timedelta(minutes=self._ON_DEMAND_REFRESHING_INTERVAL_MINUTES)
+        return self._token.expires_on - self._get_utc_now_as_int() <\
+               timedelta(minutes=self._ON_DEMAND_REFRESHING_INTERVAL_MINUTES)
 
     def _is_currenttoken_valid(self):
-        return self._get_utc_now() < self._token.expires_on
+        return self._get_utc_now_as_int() < self._token.expires_on
 
     @classmethod
-    def _get_utc_now(cls):
-        return datetime.now().replace(tzinfo=TZ_UTC)
+    def _get_utc_now_as_int(cls):
+        current_utc_datetime = datetime.now().replace(tzinfo=TZ_UTC)
```

Comment:
Maybe this is a good time to check whether this is supposed to be `datetime.utcnow()`?

Reply:
changed it to utc time